skyportal/handlers/api/comment.py

```diff
@@ -59,32 +71,71 @@ def get(self, comment_id, associated_resource_type=None):
                 schema: Error
         """
 
-        if associated_resource_type is None:
-            associated_resource_type = 'object'
-
-        if associated_resource_type.lower() == "object":  # comment on object (default)
+        try:
+            comment_id = int(comment_id)
+        except (TypeError, ValueError):
+            return self.error("Must provide a valid comment ID. ")
+
+        # the default is to comment on an object
+        if associated_resource_type.lower() in (
+            "object",
+            "objects",
+            "source",
+            "sources",
+        ):
             comment = Comment.get_if_accessible_by(
                 comment_id, self.current_user, raise_if_none=True
             )
-        elif associated_resource_type.lower() == "spectrum":
+            comment_resource_id_str = str(comment.obj_id)
+
+            # comment.api_path = f'api/sources/{resource_id}/comment/{comment_id}'
+        elif associated_resource_type.lower() in ("spectrum", "spectra"):
             comment = CommentOnSpectrum.get_if_accessible_by(
                 comment_id, self.current_user, raise_if_none=True
             )
+            comment_resource_id_str = str(comment.spectrum_id)
+            # comment.api_path = f'api/spectrum/{resource_id}/comment/{comment_id}'
         # add more options using elif
         else:
             return self.error(
                 f'Unsupported input "{associated_resource_type}" given as "associated_resource_type" argument.'
             )
 
+        if comment_resource_id_str != resource_id:
+            return self.error(
+                f'Comment resource ID does not match resource ID given in path ({resource_id})'
+            )
         return self.success(data=comment)
 
     @permissions(['Comment'])
-    def post(self):
+    def post(self, associated_resource_type, resource_id, *_):
```

Comment:
When do more keyword arguments land here?

Reply:
This fits the signature from the tornado server regexp.

Reply:
I guess I'm asking when the `*_` part is used.

Reply:
It is never used, but if I don't add that last input it complains about not having the right number of arguments.

Reply:
That's not right; for the post there should not be a comment ID there.  A comment ID is only provided for requests that work with an existing comment.

Can you confirm that this is problematic when posting to: `/api/sources/ztf123/comment` ?

Reply:
When I remove `*_` as the fourth argument, I get this error when posting from the front-end:

`post() takes 3 positional arguments but 4 were given`

Reply:
The last argument is "None" but it still exists...

Reply:
The regexp for the server is `r'/api/(sources|spectrum)/([0-9A-Za-z-_\.\+]+)/comment(/[0-9]+)?'` which transmits that last input argument even if the last group doesn't contain anything.

Reply:
I can solve this by adding a `(r'/api/(sources|spectrum)/([0-9A-Za-z-_\.\+]+)/comment', CommentHandler)` route. I don't know why I didn't come up with that earlier...

Reply:
Thanks for the explanation; I understand now.  I also found a SO post that discusses the same issue: https://stackoverflow.com/a/58775110/214686

I think your solution (two routes) is better.  It's a pity we won't easily be able to print out an error message for the case where that argument *is* supplied, but I'm not sure how to solve that simply.  If users follow the API docs, they won't run into the issue, so should be OK.

Reply:
Minor point: the plural of spectrum is spectra. Do we want to harmonize the plural/singular with `sources` in the API call?

Reply:
Yes, we've been wanting to do that for a while.  See https://github.com/skyportal/skyportal/issues/610

I think we should make things consistent, with deprecation aliases in place (in another PR).

Reply:
I can easily put both spectra and spectrum in the app_server.py and allow the comments/annotations handlers to comply with both, and we can deprecate `spectrum` at some later time. 

What do you think? @profjsb @stefanv

Reply:
As discussed this morning, please keep PRs to features implemented right now, and the minimal changes necessary for that.

Reply:
OK, in that case the PR is basically ready for a final round of reviews (I hope)